sympy/core/power.py

```diff
@@ -582,6 +582,9 @@ def _check(ct1, ct2, old):
                     combines = Pow._eval_power(
                         Pow(*old.as_base_exp(), evaluate=False),
                         pow) is not None
+                if not combines:
+                    if pow.has(pi):
```

Comment:
I'm unclear on exactly what it is you're trying to do, but be aware that `x**y**z != x**(y*z)` in general.